src/doc/book/unsized-types.md

```diff
@@ -47,14 +47,13 @@ pointers, can use this `impl`.
 # ?Sized
 
 If you want to write a function that accepts a dynamically sized type, you
-can use the special bound, `?Sized`:
+can use the special syntax, `?Sized`:
 
 ```rust
 struct Foo<T: ?Sized> {
     f: T,
 }
 ```
 
-This `?`, read as “T may be `Sized`”,  means that this bound is special: it
-lets us match more kinds, not less. It’s almost like every `T` implicitly has
-`T: Sized`, and the `?` undoes this default.
+This `?Sized`, read as “T may or may not be `Sized`”, allowing us to match both constant size and unsized types.
+All generic type parameters implicitly have the `Sized` bound, so `?Sized` can be used to opt-out of the implicit bound.
```

Comment:
Small nit, the line wrapping here is different to the rest of the file, it should probably be whatever the rest of the text is (looks like 80 chars?).